chapters/20-nodejs.md

```diff
@@ -491,3 +491,61 @@ Objetos que emitem eventos no Node têm um método chamado `on` que é similar
 ao método `addEventListener` no navegador. Você dá um nome de evento e então
 uma função, e isso irá registrar uma função para ser chamada toda vez que um
 dado evento ocorrer.
+
+_Streams_ de leitura possuem os eventos `"data"` e `"end"`. O primeiro é
+acionado sempre que existe alguma informação chegando, e o segundo é chamado
+sempre que a _stream_ chega ao fim. Esse modelo é mais adequado para um
+_streamming_ de dados, que pode ser imediatamente processado, mesmo quando todo
+documento ainda não está disponíevel. Um arquivo pode ser lido como uma _stream_
+de leitura usando a função `fs.createReadStream`.
+
+O seguinte código  cria um servidor que lê o corpo da requisição e o devolve em
+caixa alta para o cliente via _stream_:
+
+```javascript
+var http = require("http");
+http.createServer(function(request, response) {
+  response.writeHead(200, {"Content-Type": "text/plain"});
+  request.on("data", function(chunk) {
+    response.write(chunk.toString().toUpperCase());
+  });
+  request.on("end", function() {
+    response.end();
+  });
+}).listen(8000);
+```
+
+A variável `chunk` enviada para o manipulador de dados será um `Buffer` binário,
+que nós podemos converter para uma _string_ chamando `toString` nele, que vai
+decodificá-lo usando a codificação padrão (UTF-8).
```

Comment:
Vamos dar uma olhada no texto original:

> The chunk variable passed to the data handler will be a binary Buffer, which we can convert to a string by calling toString on it, which will decode it using the default encoding (UTF-8).

Acredito que quando ele fala _“[…] which will decode it using the default encoding (UTF-8)”_, ele se refere ao fato de que quando chamamos `.toString()` em um Buffer, ele é convertido para uma String usando algum tipo de encoding.  Como o método `.toString()` pode receber diferentes tipos de encoding, e nesse exemplo não estamos informando qual o `encoding` que queremos usar, ele acaba usando o **valor (codificação) padrão** que é UTF-8.  Acredito que “UTF-8” (usado entre parênteses) é só uma referência para o leitor saber que o valor “default” nesse caso é o UTF-8.  Podemos verificar olhando a documentação:

![](https://s3.amazonaws.com/f.cl.ly/items/3h012e0P280P3L08082Q/Screen%20Shot%202015-03-30%20at%2012.00.21%20PM.png)
_https://nodejs.org/api/buffer.html#buffer_buf_tostring_encoding_start_end_

Se tivermos sentimentos não conclusivos quanto a isso, podemos abrir uma issue para termos mais perspectivas, o que acham?